### PR TITLE
[SYCLomatic] Fix a crash in case for `cuEvent*` API with explicist case parameter

### PR DESCRIPTION
**Pull request by the-slow-one**
The following code would result in SIGABRT.

```cpp
#include <cstdint>
#include <cuda.h>

void foo(CUevent hEvent, CUstream hStream ) {
  CUresult result = cuEventRecord((CUevent)hEvent, (CUstream)hStream);
}
```

Issue is with the `isa<>` check of first argument's type, specifically when it's typecasted as shown in example.
Fix is to handle the explicit cast expression of the first argument to `cuEvent*` APIs.

### CODE REVIEW

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -7367,15 +7369,19 @@ void EventAPICallRule::handleEventRecordWithProfilingDisabled(
   // Insert the helper variable right after the event variables
   static std::set<std::pair<const Decl *, std::string>> DeclDupFilter;
   auto &SM = DpctGlobalInfo::getSourceManager();
-  const ValueDecl *MD = getDecl(CE->getArg(0));
-  std::string InsertStr;
+
+  const ValueDecl *MD = nullptr;
+  if((MD = getDecl(CE->getArg(0))) == nullptr)
```
Comment (the-slow-one) [inline clang/lib/DPCT/ASTTraversal.cpp]:
```suggestion
  if ((MD = getDecl(CE->getArg(0))) == nullptr)
```

File: clang/test/dpct/driver-stream-and-event.cu [hunk 2]
```diff
@@ -123,3 +123,9 @@ void test_stream() {
   // CHECK: dpct::get_current_device().destroy_queue(hStream);
   cuStreamDestroy(hStream);
 }
+
+void test_cuEventRecord_crash(uint64_t hEvent,uint64_t hStream )
+{
+  // CHECK: int result = DPCT_CHECK_ERROR(*(CUevent)hEvent = (CUstream)hStream->ext_oneapi_submit_barrier());
```
Comment (zhiweij1) [inline clang/test/dpct/driver-stream-and-event.cu]: The type of the C-style cast also needs to be migrated.
I think it should be something like:
```
// CHECK: int result = DPCT_CHECK_ERROR(*(dpct::event_ptr)hEvent = (dpct::queue_ptr)hStream->ext_oneapi_submit_barrier());
```

**Review by zhiweij1 — CHANGES_REQUESTED** [review]

File: clang/test/dpct/driver-stream-and-event.cu (hunk 2 above)
Comment (the-slow-one) [reply clang/test/dpct/driver-stream-and-event.cu]: Done

**Review by zhiweij1 — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Pull request merged**